### PR TITLE
Implemented uploading of files to the attachmentService

### PR DESCRIPTION
**Pull request by LewisClark-HO**
### Description
This PR implements the uploading of any files added during a form. The files are uploaded to the attachment service.

As requested in [EAB-995](https://support.cop.homeoffice.gov.uk/browse/EAB-995), this PR adds `size` and `url` fields to the file data that gets submitted to the backend.

### Testing
Accompanying unit tests are included.

### CODE REVIEW

File: client/src/pages/cases/components/CaseAction.jsx [hunk 1]
```diff
@@ -345,7 +345,9 @@ const CaseAction = ({
                       form.data,
                       payload,
                       keycloak.tokenParsed.email,
-                      axiosInstance
+                      axiosInstance,
+                      fileService,
+                      contexts.data.environmentContext.attachmentServiceUrl
```
Comment (gardnersam) [inline client/src/pages/cases/components/CaseAction.jsx:350]: why is this getting changed?

File: client/src/utils/Form/uploadDocuments.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { Utils } from '@ukhomeoffice/cop-react-form-renderer';
+
+const uploadDocument = async (
+  fileServiceClient,
+  attachmentServiceURL,
+  submissionData,
+  document
+) => {
+  const formData = new FormData();
+  formData.append('file', document.file);
```
Comment (gardnersam) [inline client/src/utils/Form/uploadDocuments.js:10]: looks like this is never used again

File: client/src/utils/Form/formHooks.js [hunk 3]
```diff
@@ -1,23 +1,36 @@
...
+  // This has to be done before adjustPayloadForType is
+  // called as that wipes out any meta.document data on
+  // the submission payload.
+  if (fileServiceClient && attachmentServiceURL) {
+    await uploadDocuments(fileServiceClient, attachmentServiceURL, submissionPayload);
+  }
+  const adjustedPayload = adjustPayloadForType(type, form.name, submissionPayload);
+  const { data } = await axiosFn(submissionUrl, adjustedPayload);
+  return { data, businessKey: submissionPayload.businessKey };
 };
```
Comment (gardnersam) [inline client/src/utils/Form/formHooks.js:34]: the tests for this function should be updated

File: client/src/utils/Form/uploadDocuments.test.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+          { id: 'a', url: 'http://file.com/thing.jpg', field: 'alpha' },
+          { id: 'b', file: { bravo: 'bcd' }, field: 'bravo' },
+          { id: 'c', file: { charlie: 'cde' }, field: 'charlie' },
+        ],
+      },
+    };
+    const result = await uploadDocuments(EMPTY_CLIENT, '/files/TEST', PAYLOAD);
+    expect(Array.isArray(result)).toBeTruthy();
+    expect(result[0]).toMatchObject(PAYLOAD.meta.documents[1]);
+    expect(result[1]).toMatchObject(PAYLOAD.meta.documents[2]);
```
Comment (gardnersam) [inline client/src/utils/Form/uploadDocuments.test.js:102]: Your usage of the function ignores the return value and relies on the submissionPayload getting updated but these tests are mostly all focused on the return value. Could they be extended to check that PAYLOAD has or hasn't been updated appropriately?

Comment (LewisClark-HO) [reply client/src/utils/Form/uploadDocuments.test.js:102]: Sorry not sure what you mean here, `result` is used in both tests - where abouts do you think it's being ignored? If `uploadDocuments` receives an empty response from the client then it should just return the original file data, that's why I'm checking against `PAYLOAD`.

Comment (LewisClark-HO) [reply client/src/utils/Form/uploadDocuments.test.js:102]: Ahh gotcha, good shout

**Review by gardnersam — APPROVED** [review]
LGTM

**Pull request merged**